src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1735,15 +1780,23 @@ class Typer extends Namer with TypeAssigner with Applications with Implicits wit
           }
           tree
         }
-        else if (tree.tpe <:< pt)
-          if (ctx.typeComparer.GADTused && pt.isValueType)
+        else if (tree.tpe <:< pt) {
+          if (pt.hasAnnotation(defn.InlineParamAnnot))
+            checkInlineConformant(tree, "argument to inline parameter")
+          if (Inliner.hasBodyToInline(tree.symbol) &&
+              !ctx.owner.ownersIterator.exists(_.isInlineMethod) &&
+              !ctx.settings.YnoInline.value &&
+              !ctx.isAfterTyper)
+            adapt(Inliner.inlineCall(tree, pt), pt)
```

Comment:
This is the only call to `inlineCall` and it's in a branch with `tree.tpe <:< pt` this means that when `adaptToSubType` is necessary we never inline, for example `get` isn't inlined in `foo`:
```scala
object Test {
  inline def get: Int = 1

  def foo: Unit = {
    get
  }
}
```